### PR TITLE
chore: Update golangci-linter config and clean repo

### PR DESCRIPTION
**Pull request by kaymckay**
This PR adds the new golangci linter configuration and cleans the repo. Since there was a lot, I cleaned a1-2 linters per commit for better visibility of the changes.

some things to note:
- noctx yelled at not having contexts with requests, so context.background() was added
- middleware/token.go uses a deprecated [StandardClaims](https://pkg.go.dev/github.com/golang-jwt/jwt/v4@v4.3.0#StandardClaims) which is currently linted as a false positive, but we may want to upgrade to [RegisterClaims](https://pkg.go.dev/github.com/golang-jwt/jwt/v4@v4.3.0#RegisteredClaims) -> causes some typing issues
  - will make an issue for this
- removed changelog code

### CODE REVIEW

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Review by wass3r — COMMENTED** [review]
some minor comments - let me know what you think. thanks for all the work!

File: api/build.go [hunk 1]
```diff
@@ -118,11 +117,8 @@ func CreateBuild(c *gin.Context) {
 	}
 
 	// verify the build has a valid event and the repo allows that event type
-	if (input.GetEvent() == constants.EventPush && !r.GetAllowPush()) ||
-		(input.GetEvent() == constants.EventPull && !r.GetAllowPull()) ||
-		(input.GetEvent() == constants.EventTag && !r.GetAllowTag()) ||
-		(input.GetEvent() == constants.EventDeploy && !r.GetAllowDeploy()) {
-		// nolint: lll // ignore long line length due to error message
+	if (input.GetEvent() == constants.EventPush && !r.GetAllowPush()) || (input.GetEvent() == constants.EventPull && !r.GetAllowPull()) ||
+		(input.GetEvent() == constants.EventTag && !r.GetAllowTag()) || (input.GetEvent() == constants.EventDeploy && !r.GetAllowDeploy()) {
```
[the remark this answers is not in the document]
Comment (kaymckay) [reply api/build.go:121]: Nope it can be switched back 👍

File: api/webhook.go [hunk 2]
```diff
@@ -246,12 +249,9 @@ func PostWebhook(c *gin.Context) {
 	}
 
 	// verify the build has a valid event and the repo allows that event type
-	if (b.GetEvent() == constants.EventPush && !r.GetAllowPush()) ||
-		(b.GetEvent() == constants.EventPull && !r.GetAllowPull()) ||
-		(b.GetEvent() == constants.EventComment && !r.GetAllowComment()) ||
-		(b.GetEvent() == constants.EventTag && !r.GetAllowTag()) ||
+	if (b.GetEvent() == constants.EventPush && !r.GetAllowPush()) || (b.GetEvent() == constants.EventPull && !r.GetAllowPull()) ||
+		(b.GetEvent() == constants.EventComment && !r.GetAllowComment()) || (b.GetEvent() == constants.EventTag && !r.GetAllowTag()) ||
```
[the remark this answers is not in the document]
Comment (kaymckay) [reply api/webhook.go:253]: 👍

File: cmd/vela-server/main.go [hunk 3]
```diff
@@ -155,53 +148,40 @@ func main() {
...
 			EnvVars: []string{"VELA_WORKER_ACTIVE_INTERVAL", "WORKER_ACTIVE_INTERVAL"},
 			Name:    "worker-active-interval",
 			Usage:   "interval at which workers will show as active within the /metrics endpoint",
 			Value:   5 * time.Minute,
 		},
 	}
-
-	// Database Flags
-
+	// Add Database, Queue, Secret, and Source Flags
```
[the remark this answers is not in the document]
Comment (kaymckay) [reply cmd/vela-server/main.go:175]: It was to keep main < 160 lines long, but it can be switched back & false pos

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Review by wass3r — APPROVED** [review]

**Pull request merged**